INTMDB-263: Create Resource and Datasource for private_link_endpoint_adl (#640)
mongodb/terraform-provider-mongodbatlas: website/docs/d/privatelink_endpoint_service_adl.html.markdown

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: privatelink_endpoint_service_adl"
sidebar_current: "docs-mongodbatlas-datasource-privatelink-endpoint-service-adl"
description: |-

> Describes a DataLake and Online Archive private link endpoint.

Reviewer: ```suggestion
Describes an Atlas Data Lake and Online Archive PrivateLink endpoint
```

Author: done!

---

# privatelink_endpoint_service_adl

> `privatelink_endpoint_service_adl` Provides DataLake and Online Archive private link endpoint resource.

Reviewer: ```suggestion
`privatelink_endpoint_service_adl` Provides an Atlas Data Lake (ADL) and Online Archive PrivateLink endpoint resource.
```

Author: done!

-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.

## Example Usage

### Basic
```terraform
resource "mongodbatlas_privatelink_endpoint_service_adl" "adl_test" {
  project_id   = "<PROJECT_ID>"
  endpoint_id  = "<ENDPOINT_ID>"

>   comment      = "comments for private link endpoint adl"

Reviewer: ```suggestion
  comment      = "Comment for PrivateLink endpoint ADL"
```

Author: done!

  type		 = "DATA_LAKE"
  provider_name	 = "AWS"
}

data "mongodbatlas_privatelink_endpoint_service_adl" "test" {
  project_id            = mongodbatlas_privatelink_endpoint_service_adl.adl_test.project_id
  private_link_id       = mongodbatlas_privatelink_endpoint_service_adl.adl_test.endpoint_id
}
```

## Argument Reference

* `project_id` - (Required) Unique 24-digit hexadecimal string that identifies the project.
* `endpoint_id` - (Required) Unique 22-character alphanumeric string that identifies the private endpoint. Atlas supports AWS private endpoints using the [|aws| PrivateLink](https://aws.amazon.com/privatelink/) feature.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

> * `type` - Human-readable label that identifies the resource associated with this private endpoint. Value is `DATA_LAKE`.

Reviewer: ```suggestion
* `type` - Human-readable label that identifies the type of resource to associate with this private endpoint.
```

Author: done!

> * `provider_name` - Human-readable label that identifies the cloud provider for this endpoint. Value is AWS.

Reviewer: ```suggestion
* `provider_name` - Human-readable label that identifies the cloud provider for this endpoint. 
```

Author: done!
